File: Semester 1/Class11_Computer_Contest_Lv_2_Notes_and_Homework_Nov_20-21/lvl2_hw_sol/lvl2_boxes.cpp
#include <iostream>
#include <algorithm>
using namespace std;
struct box{
    int l, w, h, volume;
}boxes[101], items[101];
int n, m, a[3];

bool cmp(box x, box y){
    return x.volume < y.volume;
}

int main(){
    cin >> n;
    for(int i=0; i<n; i++){
        cin >> a[0] >> a[1] >> a[2];
        sort(a, a+3);
        boxes[i].l=a[0]; boxes[i].w=a[1]; boxes[i].h=a[2];
        boxes[i].volume = a[0]*a[1]*a[2];
    }
    sort(boxes, boxes+n, cmp);

    cin >> m;
    for(int i=0; i<m; i++){
        cin >> a[0] >> a[1] >> a[2];
        sort(a, a+3);
        items[i].l=a[0]; items[i].w=a[1]; items[i].h=a[2];
        items[i].volume = a[0]*a[1]*a[2];
    }

    for(int i=0; i<m; i++){
        bool output = false;
        for(int j=0; j<n; j++){
            if (items[i].l<=boxes[j].l && items[i].w<=boxes[j].w
                && items[i].h<=boxes[j].h){
                cout << boxes[j].volume << endl;
                output = true;
                break;
            }
        }
        if(!output)
            cout << "Item does not fit." << endl;
    }
    return 0;
}

File: Semester 1/Class11_Computer_Contest_Lv_2_Notes_and_Homework_Nov_20-21/lvl2_hw_sol/lvl2_boxes.java
import java.util.*;
public class homework {
	public static void main (String [] args){
		Scanner in = new Scanner(System.in);
		int n = in.nextInt(), box[][] = new int[n][3];
		for(int i=0; i<n; i++) {
			for(int j=0; j<3; j++)
				box[i][j] = in.nextInt();
			Arrays.sort(box[i]);
		}
		int m = in.nextInt(), item[] = new int[3];
		for(int k=0; k<m; k++) {
			for(int j=0; j<3; j++)
				item[j] = in.nextInt();
			Arrays.sort(item);
			long vol = Long.MAX_VALUE;
			for(int i=0; i<n; i++) {
				if(box[i][0] >= item[0] && box[i][1] >= item[1] && box[i][2] >= item[2])
					vol = Math.min(vol, (long)box[i][0] * box[i][1] * box[i][2]);
			}
			if(vol == Long.MAX_VALUE) System.out.println("Item does not fit.");
			else System.out.println(vol);
		}
	}
}